pandas/core/algorithms.py

```diff
@@ -1575,7 +1578,11 @@ def take_nd(arr, indexer, axis=0, out=None, fill_value=np.nan, mask_info=None,
     # TODO(EA): Remove these if / elifs as datetimeTZ, interval, become EAs
     # dispatch to internal type takes
     if is_extension_array_dtype(arr):
-        return arr.take(indexer, fill_value=fill_value, allow_fill=allow_fill)
+        try:
+            return arr.take(indexer, fill_value=fill_value, allow_fill=allow_fill, axis=axis)
+        except TypeError:
```

Comment:
I worry a bit about `axis` not being the only reason for a TypeError. Can you check `_allows_2d` instead?

Reply:
Good idea.